packages/suite/src/hooks/wallet/coinmarket/useCoinmarketWatchTrade.ts

```diff
@@ -0,0 +1,130 @@
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useTimeoutFn, useUnmount } from 'react-use';
+import { Trade } from 'src/types/wallet/coinmarketCommonTypes';
+import { useFormDraft } from '../useFormDraft';
+import invityAPI from 'src/services/suite/invityAPI';
+import { saveTrade as saveBuyTrade } from 'src/actions/wallet/coinmarketBuyActions';
+import { saveTrade as saveExchangeTrade } from 'src/actions/wallet/coinmarketExchangeActions';
+import { saveTrade as saveSellTrade } from 'src/actions/wallet/coinmarketSellActions';
+import {
+    WatchBuyTradeResponse,
+    WatchExchangeTradeResponse,
+    WatchSellTradeResponse,
+} from 'invity-api';
+import {
+    CoinmarketTradeStatusType,
+    CoinmarketTradeType,
+    CoinmarketUseWatchTradeProps,
+    CoinmarketWatchTradeProps,
+} from 'src/types/coinmarket/coinmarket';
+
+export const getTradeFinalStatuses = (tradeType: string): Partial<CoinmarketTradeStatusType>[] => {
+    if (tradeType === 'buy') {
+        return ['SUCCESS', 'ERROR', 'BLOCKED'];
+    }
+
+    if (tradeType === 'sell') {
+        return ['SUCCESS', 'ERROR', 'BLOCKED', 'CANCELLED', 'REFUNDED'];
+    }
+
+    if (tradeType === 'exchange') {
+        return ['SUCCESS', 'ERROR', 'KYC'];
+    }
```

Comment:
```suggestion
export const getTradeFinalStatuses = (tradeType: TradeType): Partial<CoinmarketTradeStatusType>[] => {
    if (tradeType === 'buy') {
        return ['SUCCESS', 'ERROR', 'BLOCKED'] satisfies BuyTradeFinalStatus[];
    }

    if (tradeType === 'sell') {
        return ['SUCCESS', 'ERROR', 'BLOCKED', 'CANCELLED', 'REFUNDED'] satisfies SellTradeFinalStatus[];
    }

    if (tradeType === 'exchange') {
        return ['SUCCESS', 'ERROR', 'KYC'] satisfies ExchangeTradeFinalStatus[];
    }
```

nit: is function necessary? how about static object?
```typescript
export const tradeFinalStatuses: Record<TradeType, CoinmarketTradeStatusType[]> = {
    buy: ['SUCCESS', 'ERROR', 'BLOCKED'] satisfies BuyTradeFinalStatus[],
    sell: ['SUCCESS', 'ERROR', 'BLOCKED', 'CANCELLED', 'REFUNDED'] satisfies SellTradeFinalStatus[],
    exchange: ['SUCCESS', 'ERROR', 'KYC'] satisfies ExchangeTradeFinalStatus[],
    savings: [],
    spend: [],
};
```